### DISCUSSION TITLE
AP CERASMARTER is not starting

### DISCUSSION THREAD

**Discussion by falcke** [Neuroquila-n8fall/JunkersControl#43] [Q&A]
Hello,
I just began to try out this project and would like to start with a big thank you!
Following the [Installation (Quick Start)](https://github.com/Neuroquila-n8fall/JunkersControl/discussions/new?category=q-a#installation-quick-start) I've flashed an ESP32-WROOM-32 dev board with the [precompiled binaries of the latest version](https://github.com/Neuroquila-n8fall/JunkersControl/releases/tag/v0.9.3-3) by just and only using the esptool as follows.

_firmware.bin_
```
esptool.py --after no_reset --chip esp32 --baud 921600 --port /dev/cu.wchusbserial10 write_flash --verify 0x10000 firmware.bin 
esptool.py v4.6.2
Serial port /dev/cu.wchusbserial10
Connecting.....
Chip is ESP32-D0WD-V3 (revision v3.1)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: 08:d1:f9:e8:40:34
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 921600
Changed.
Configuring flash size...
Flash will be erased from 0x00010000 to 0x00107fff...
Compressed 1012384 bytes to 639862...
Wrote 1012384 bytes (639862 compressed) at 0x00010000 in 8.9 seconds (effective 913.5 kbit/s)...
Hash of data verified.

Leaving...
Verifying just-written flash...
(This option is deprecated, flash contents are now always read back after flashing.)
Verifying 0xf72a0 (1012384) bytes @ 0x00010000 in flash against firmware.bin...
-- verify OK (digest matched)
Staying in bootloader.

```

_littlefs.bin_
```
esptool.py --after hard_reset --chip esp32 --baud 921600 --port /dev/cu.wchusbserial10 write_flash 0x307000 littlefs.bin
esptool.py v4.6.2
Serial port /dev/cu.wchusbserial10
Connecting......
Chip is ESP32-D0WD-V3 (revision v3.1)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: 08:d1:f9:e8:40:34
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 921600
Changed.
Configuring flash size...
Flash will be erased from 0x00307000 to 0x003f6fff...
Compressed 983040 bytes to 366828...
Wrote 983040 bytes (366828 compressed) at 0x00307000 in 6.0 seconds (effective 1300.9 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...
```

So far the console output doesn't look faulty to me but I don't get the output shown in the documentation

> Press the "BOOT" button within the next 5 seconds to enable Setup Mode!
Setup Mode not enabled. You can enable it at every time by pressing the "BOOT" button once.
[  5047][E][vfs_api.cpp:104] open(): /littlefs/configuration.json does not exist, no permits for creation
Configuration file could not be found. Please upload it first.
Unable to read configuration.
Invalid WiFi configuration. Launching AP mode.
WiFi AP launched. Find me @ 192.168.4.1
Can't connect to MQTT broker. [No Network]
Can't connect to MQTT broker. [No Network]
Can't connect to MQTT broker. [No Network]
Can't connect to MQTT broker. [No Network]
Can't connect to MQTT broker. [No Network]

the AP also won't start or show off. Pressing the Boot button won't help here.
Am I missing anything?
Thank you!
Marcel

**Comment by BernhardS87** [comment]
@falcke Did you find a solution? I have exactly the same problem. There is a WIFI called CERASMARTER, I can also connect to it but I can't connect to http://cerasmarter. The page is not found.

**Reply by falcke** [reply]
@BernhardS87 No. I'm stuck at the exact same spot as you are. Flashing by only using the esptool didn't even bring me so far. Using VS Code and platform.io took me where I'm at right now.
Any advice is highly appreciated ;)

**Reply by BernhardS87** [reply]
@falcke I think I made a mistake. I started with VS code and platformio and had some problems but it was obviously working nevertheless (but I didnt realize it). Then I was trying esptool and saw that there is the CERASMARTER wifi and thought it was the result of the esptool upload. After resetting the esp32 I was trying it again and figured out only the upload with VC code and platformio is resulting in the CERASMARTER network. 

Did you modify the configuration.json file and copied it into the data folder before starting the upload with platformio? I only had to change the upload port in the platformio file in order to upload it successfully. I think there is still an issue that the esp32 can't connect to the WIFI because I can't see a new device in my network.

**Reply by falcke** [reply]
@BernhardS87 I tried it either way. I'm still stuck with being able to connect via the AP CERASMARTER but I won't be able to connect to the web ui.

**Reply by BernhardS87** [reply]
@falcke Somehow I get now the following message. I was using the "serial monitor" button before

`Running Environment: production
Running Build: "LOCAL_BUILD"
Press the "BOOT" button within the next 5 seconds to enable Setup Mode!
Setup Mode not enabled. You can enable it at every time by pressing the "BOOT" button once. 
[  5043][E][vfs_api.cpp:105] open(): /littlefs/configuration.json does not exist, no permits for creation
Configuration file could not be found. Please upload it first.
Unable to read configuration.
WiFi AP launched. Find me @ 192.168.4.1`

I am able to ping the address but can't connect to webUI. @Neuroquila-n8fall  could you maybe give us some hints?

**Reply by Neuroquila-n8fall** [reply]
As answered below you should be able to access the UI at http://192.168.4.1

You then have the possibility to upload the configuration.json file

More often than anticipated mDNS does not work on your device (doesn't matter of it's apple, android, windows or whatever).

Are you able to access the Ui using the IP address?

**Reply by Neuroquila-n8fall** [reply]
Also you probably want to have it connect to your home network. You could just simply change this within the config file:
```json 
    "Wifi": {
        "SSID": "ssid",
        "Password": "pass",
        "Hostname": "CERASMARTER"
    },
```
... to allow it to connect to your network.

1) Change configuration.json to your likings
2) put it in the "data" folder
3) Make sure you select the "Release" profile within Platformio. If you don't do this, the "data" folder might be not properly prepared and the UI won't work anyway.
4) Upload filesystem image

**Comment by gradius-v** [comment]
I have found that setting Home Assistant to true in the configuration.json makes the web ui unreachable.

**Reply by BernhardS87** [reply]
@gradius-v but have been able to access the web ui?

**Reply by Neuroquila-n8fall** [reply]
The Home Assistant flag is for the auto setup through HA which is a feature I have not yet completed (because it is freakin tedious)

Unintentional side effects are very likely ;)

**Comment by Neuroquila-n8fall** [comment]
Sorry all for the late reply. I was busy with other things.

So first and foremost if you connect to the AP you should pay attention to the IP address the controller gives you via serial output since the "mDNS" service may not work always. 
So "http://cerasmarter" might not work for you - try http://192.168.4.1 instead.

By the looks of it (serial output) you did everything correct.

**Reply by BernhardS87** [reply]
Thanks for the reply. I am able to ping the 192.168.4.1 once I am connected to the CERASMARTER wifi but I am not able to access the UI even when using the IP address.

**Reply by Neuroquila-n8fall** [reply]
What is the serial output saying while you attempt to access the UI? I think something might be off with the filesystem image so that it can't load and transmit the UI.
